src/cpp/src/tokenizer.cpp

```diff
@@ -500,6 +502,11 @@ std::string Tokenizer::apply_chat_template(ChatHistory history,
     return m_pimpl->apply_chat_template(history, add_generation_prompt, chat_template);
 }
 
+bool Tokenizer::is_chat_template_ready() const {
+    return m_pimpl->is_chat_template_ready();
+};
```

Comment:
There's no reason to expose this outside. Throw from `apply_chat_template()`.

Reply:
done